service/storage/operations_internal_test.go

```diff
@@ -1255,3 +1256,113 @@ func TestIndexAndLookup_Seals(t *testing.T) {
 		assert.Equal(t, 1, decodeCallCount)
 	})
 }
+
+func TestLibrary_IterateLedger(t *testing.T) {
+	entries := 5
+	paths := mocks.GenericLedgerPaths(entries)
+	payloads := mocks.GenericLedgerPayloads(entries)
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		codec := zbor.NewCodec()
+		l := &Library{codec}
+
+		for i := 0; i < entries; i++ {
+			require.NoError(t, db.Update(l.SavePayload(mocks.GenericHeight, paths[i], payloads[i])))
+		}
+
+		var callCount int
+		op := l.IterateLedger(func(path ledger.Path, payload *ledger.Payload) error {
+			callCount++
+
+			assert.Contains(t, paths, path)
+			assert.Contains(t, payloads, payload)
```

Comment:
I think this test is important enough to make sure that we actually iterate through each path and payload exactly once. So instead of an overall call count, you could have a map with call counts mapped by ledger path, and make sure each one is traversed exactly once.